run-tests.sh

```diff
@@ -89,6 +90,26 @@ for arch in $ARCHS; do
             ;;
         esac
     done
+    for test in $TESTS_WINRT; do
+        set +e
+        # compilation should fail for WinRT
+        $arch-w64-$target_os-clang $test.c -o $TEST_DIR/$test.exe -Wimplicit-function-declaration -Werror
+        WINRT_ERROR=$?
+        set -e
+        case $target_os in
+        mingw32winrt)
+            if [ $WINRT_ERROR -eq 0 ]; then
+                echo "WinRT compilation should have failed for test $test!"
+                exit 1
+            fi
+            ;;
+        *)
+            if [ $WINRT_ERROR -ne 0 ]; then
```

Comment:
Same here, should error out if this testcase failed building for UWP/desktop, and run it if `$WINRT_ERROR -eq 0`.